source/fundamentals/monitoring.txt

```diff
@@ -0,0 +1,411 @@
+- Add an instance of the new class that implements ``CommandListener`` to a ``MongoClientSettings`` object.
+- Configure a ``MongoClient`` instance with the ``MongoClientSettings`` object.
+
+The following code defines the ``CommandTimer`` class which implements the
+``CommandListener`` interface:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Monitoring.java
+   :language: java
+   :dedent:
+   :start-after: start command-listener-impl
+   :end-before: end command-listener-impl
+
+The following code adds an instance of the ``CommandTimer`` class to a
+``MongoClientSettings`` object, and configures a ``MongoClient`` instance with the
+``MongoClientSettings`` object. The code then runs some database commands to test the
+timer.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Monitoring.java
+   :language: java
+   :dedent:
+   :start-after: start monitor-command-example
+   :end-before: end monitor-command-example
+
+The above code snippet should produce output that looks like this:
+
+.. code-block:: none
+   :copyable: false
+
+   Command 'find' took 19 milliseconds
+   Command 'aggregate' took 15 milliseconds
+   Command 'endSessions' took 14 milliseconds
+
+For more information on the classes and methods mentioned in this section, see
+the following API documentation:
+
+- :java-docs:`CommandListener <apidocs/mongodb-driver-core/com/mongodb/event/CommandListener.html>`
+- :java-docs:`MongoClientSettings <apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.html>`
+- :java-docs:`MongoClient <apidocs/mongodb-driver-sync/com/mongodb/client/MongoClient.html>`
+- :java-docs:`CommandStartedEvent <apidocs/mongodb-driver-core/com/mongodb/event/CommandStartedEvent.html>`
+- :java-docs:`CommandSucceededEvent <apidocs/mongodb-driver-core/com/mongodb/event/CommandSucceededEvent.html>`
+- :java-docs:`CommandFailedEvent <apidocs/mongodb-driver-core/com/mongodb/event/CommandFailedEvent.html>`
+
+Cluster Events
+~~~~~~~~~~~~~~
+
+A cluster event is an event related to a change in the state of the
+MongoDB instance or cluster you have connected the driver to. Cluster events are
+also called server discovery and monitoring (SDAM) events. 
+ 
+The driver defines nine cluster events. The driver divides these nine events
+between three separate listener interfaces which each listen for three of the
+nine events. Here are the three interfaces and the events they listen for:
+
+- ``ClusterListener``: :spec:`topology </server-discovery-and-monitoring/server-discovery-and-monitoring.rst#topology>`
+  related events
+- ``ServerListener``: events related to ``mongod`` or ``mongos`` processes
+- ``ServerMonitorListener``: heartbeat related events
+
+To monitor one of the three types of cluster events, write a class that
+implements one of the three interfaces above and register an instance of that
+class with your ``MongoClient`` instance.
+
+For a detailed description of all nine cluster events in the driver, see the 
```

Comment:
Nit: Didn't think you needed to mention nine here since you did above.
```suggestion
For a detailed description for each cluster event in the driver, see the 
```

Reply:
I agree "of all nine" can be removed here.

```suggestion
For a detailed description of each cluster event in the driver, see the 
```